plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageDatabaseUserInterface.java

```diff
@@ -177,7 +235,16 @@ public boolean performOk()
         store.setValue(DBeaverPreferences.AGENT_LONG_OPERATION_TIMEOUT, longOperationsTimeout.getSelection());
 
         PrefUtils.savePreferenceStore(store);
-
+        if (clientTimezone.getSelectionIndex() >= 0) {
+
+            if (!clientTimezone.getText().equals("DEFAULT")) {
+                store.setValue(DBeaverPreferences.CLIENT_TIMEZONE, TimezoneRegistry.extractTimezoneId(clientTimezone.getText()));
+                TimeZone.setDefault(TimeZone.getTimeZone(TimezoneRegistry.extractTimezoneId(clientTimezone.getText())));
+            } else {
+                store.setValue(DBeaverPreferences.CLIENT_TIMEZONE, clientTimezone.getText());
+                TimeZone.setDefault(null);
```

Comment:
I'm starting to think that this is what the introduced `TimezoneRegistry` should be used for.